packages/tinybench-plugin/src/instrumented.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Measurement,
+  mongoMeasurement,
+  optimizeFunction,
+  setupCore,
+  teardownCore,
+} from "@codspeed/core";
+import { Bench } from "tinybench";
+import { taskUriMap } from "./index";
+
+declare const __VERSION__: string;
+
+function getTaskUri(
+  bench: Bench,
+  taskName: string,
+  rootCallingFile: string
+): string {
+  const uriMap = taskUriMap.get(bench);
+  return uriMap?.get(taskName) || `${rootCallingFile}::${taskName}`;
+}
+
+export function runInstrumentedBench(
+  bench: Bench,
+  rootCallingFile: string
+): void {
+  bench.run = async () => {
+    console.log(
+      `[CodSpeed] running with @codspeed/tinybench v${__VERSION__} (instrumented mode)`
+    );
+    setupCore();
+
+    for (const task of bench.tasks) {
+      const uri = getTaskUri(bench, task.name, rootCallingFile);
+
+      // Access private fnOpts to get hooks
+      const fnOpts = (task as any).fnOpts;
```

Comment:
The following is better, we can still use the correct typing

```suggestion
      // Access private fields
      const { fnOpts, fn } = task as unknown as { fnOpts?: FnOptions; fn: Fn };
```